CRM/MembershipExtras/Hook/PostProcess/RecurringContributionLineItemCreator.php

```diff
@@ -103,11 +139,18 @@ private function createRecurLineItem($lineItemParams) {
     CRM_MembershipExtras_BAO_ContributionRecurLineItem::create([
       'contribution_recur_id' => $this->recurContributionID,
       'line_item_id' => $lineItemCopy['id'],
-      'start_date' => $this->recurContribution['start_date'],
+      'start_date' => $lineStartDate,
       'auto_renew' => $autoRenew,
     ]);
   }
 
+  private function updateRecurringContribution($recurringContributionStartDate) {
```

Comment:
I don't think that this is in an alignment with what we are doing to do, the point was to separate the recur contribution/contributions from the membership dates, the recur contribution start date at the end might differ from the membership date but you here are forcing it to be the same as the earliest membership start date.

Reply:
hence that we need the recur contribution start date to equal the (earliest membership start date +- cycle day) only upon renewal or offline renewal, but it should not be the case on creation (the user/admin might supply the start date of the payment plan) but this here does not account for that.

Reply:
Sounds good.